beacon-chain/sync/validate_beacon_blocks.go

```diff
@@ -80,6 +80,7 @@ func (s *Service) validateBeaconBlockPubSub(ctx context.Context, pid peer.ID, ms
 	}
 	// Check if parent is a bad block and then reject the block.
 	if s.hasBadBlock(bytesutil.ToBytes32(blk.Block.ParentRoot)) {
+		log.Debugf("Received block with root %#x that has an invalid parent %#x", blockRoot, blk.Block.ParentRoot)
```

Comment:
Can we revert this line?

1.) It's odd to have a particular debug just for `hasBadBlock`. If we do want to keep it then we should add it for the rest of the validation step

2.) I think this was only needed for debugging a bug, this hasn't shown up in a while

3.) Maybe we also can downgrade it to trace? and something like this for all the validation steps?

Reply:
It is still relevant, users have reported seeing this log and right after their node being stuck before `resync` saves it. We still have an unresolved issue with the proposer index cache which leads to any of its descendant blocks being rejected despite all blocks being valid. Removing this debug log, makes it hard to really know what is happening.

Reply:
Great. I can help. Thank you!